File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        for num in nums:
            if num not in hashmap:
                hashmap[num] = 0
            hashmap[num] += 1
        
        buckets = [[] for _ in range(len(nums) + 1)]

        for key, freq in hashmap.items():
            buckets[freq].append(key)
        
        ans = []
        print(buckets)
        for i in range(len(buckets) - 1, 0, -1):
            for num in buckets[i]:
                ans.append(num)
                if len(ans) == k:
                    return ans
        
        return ans

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        if (n <= 2) return n;
        int k = 2;
        for (int j = 2; j < n; j++) {
            if (nums[j] != nums[k - 2]) {
                nums[k] = nums[j];
                k++;
            }
        }
        return k;
    }
};